aiidalab_widgets_base/codes.py

```diff
@@ -33,43 +33,38 @@ class CodeDropdown(ipw.VBox):
     """
     selected_code = Union([Unicode(), Instance(Code)], allow_none=True)
     codes = Dict(allow_none=True)
+    allow_hidden_codes = Bool(False)
+    allow_disabled_computers = Bool(False)
 
-    def __init__(self,
-                 input_plugin,
-                 text='Select code:',
-                 hidden_codes=False,
-                 disabled_computers=False,
-                 path_to_root='../',
-                 **kwargs):
+    def __init__(self, input_plugin, text='Select code:', path_to_root='../', **kwargs):
         """Dropdown for Codes for one input plugin.
 
         input_plugin (str): Input plugin of codes to show.
 
         text (str): Text to display before the dropdown.
-
-        hidden_codes (bool): Whether to allow selecting hidden codes.
-
-        disabled_computers (bool): Whether to allow selecting codes on disabled computers.
         """
+        self.output = ipw.Output()
 
         self.input_plugin = input_plugin
-        self.hidden_codes = hidden_codes
-        self.disabled_computers = disabled_computers
-        self.output = ipw.Output()
+
+        if 'allow_hidden_codes' in kwargs:
+            self.allow_hidden_codes = kwargs['allow_hidden_codes']
+        if 'allow_disabled_computers' in kwargs:
+            self.allow_disabled_computers = kwargs['allow_disabled_computers']
```

Comment:
What's the motivation for sneakily setting these traits as part of the object construction?

Reply:
What do you mean by "sneakily setting"? I see nothing sneaking here, honestly.

Moreover, I thin in ipywidgets similar approach is implemented:

```
import ipywidgets as ipw
s = ipw.Text(
    description="AiiDA code label:",         
    style={"description_width": "initial"}
)
display(s)


s.style = {"description_width": "200px"}
display(s)
```

Reply:
As we discussed offline, those lines are not needed at all. Traits get set automatically through the `super().__init__(**kwars)` call.